osu.Game/Screens/SelectV2/BeatmapTitleWedge.cs

```diff
@@ -246,25 +248,41 @@ private void updateDisplay()
             updateOnlineDisplay();
         }
 
+        private CancellationTokenSource? lengthBpmCancellationSource;
+
         private void updateLengthAndBpmStatistics()
         {
-            var beatmapInfo = beatmap.Value.BeatmapInfo;
+            lengthBpmCancellationSource?.Cancel();
+            lengthBpmCancellationSource = new CancellationTokenSource();
 
-            double rate = ModUtils.CalculateRateWithMods(mods.Value);
+            var token = lengthBpmCancellationSource.Token;
+
+            Task.Run(() =>
+            {
+                var beatmapInfo = beatmap.Value.BeatmapInfo;
```

Comment:
What's the slow part of this code?

Reply:
Accessing `WorkingBeatmap.Beatmap`, which waits for an asynchronous beatmap load task. Revert the commit that includes this change and try the added test.

Reply:
I've added 512460e9f7774f2c00f3345cd045e2c8105def1d to better show this